### ISSUE TITLE
[Bug]: GRPC connector conflict with grpc-netty-shaded library

### ISSUE THREAD

**Issue by jakubmalek** [strimzi/strimzi-kafka-operator#10484]
### Bug Description

The Kafka image provided by Strimzi includes several [third party](https://github.com/strimzi/strimzi-kafka-operator/blob/main/docker-images/artifacts/kafka-thirdparty-libs/3.7.1/pom.xml#L33) libraries included in the Kafka classpath. 
In my setup I'm using Kafka Connect with GRPC Kafka Connector plugin having a dependency to the `grpc-netty` library. 
The problem is that Kafka Connect has the  `grpc-netty-shaded`  included in the classpath, which causes the problems with service-loader when both `grpc-netty` and `grpc-netty-shaded` present:

> java.util.ServiceConfigurationError: io.grpc.ManagedChannelProvider: io.grpc.netty.shaded.io.grpc.netty.NettyChannelProvider not a subtype\n\tat java.base/java.util.ServiceLoader.fail(ServiceLoader.java:593)\n\tat java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1244)\n\tat java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1273)\n\tat java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1309)\n\tat java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1393)\n\tat io.grpc.ServiceProviders.loadAll(ServiceProviders.java:67)\n\tat io.grpc.ManagedChannelRegistry.getDefaultRegistry(ManagedChannelRegistry.java:101)\n\tat io.grpc.ManagedChannelProvider.provider(ManagedChannelProvider.java:43)\n\tat io.grpc.ManagedChannelBuilder.forAddress(ManagedChannelBuilder.java:44)\n\tat
> ...

As a workaround, I was able to resolve the problem by excluding GRPC libs present in Kafka Connect classpath from the GRPC Connector plugin distribution package, and downgrading GRPC version used by connector from version 1.62.2 to version 1.61.0.

In general adding a third-party libraries to the main classpath of Kafka Connect can lead to problems with connector plugins, that normally are loaded in a separated classpath.
Here as I understand, the GRPC library is needed for the OpenTelemetry agent, which is disabled by default.

Perhaps instead of pre-installing OpenTelemetry dependencies into `/opt/kafka/libs` directory, the dependencies can be placed into a separate directory. 
Then depending on whether the OpenTelemetry is enabled or not, those libraries can be added to the classpath in the run script.

### Expected behavior

The OpenTelemetry dependencies are only added to the classpath when OpenTelemetry is enabled.

### Strimzi version

0.42.0

### Kubernetes version

1.29.4

### Installation method

Helm chart

### Infrastructure

Azure

**Comment by scholzj** [comment]
OpenTelemetry support (as well as the other 3rd-party libraries) is an integral part of Strimzi. So I'm not sure if we are going to remove it from the classpath. There is an issue for better separating the classpaths: https://github.com/strimzi/strimzi-kafka-operator/issues/7105 - but I do not think anyone had time to really work on it.

**Comment by jakubmalek** [comment]
I see the #7105 would indeed be a solution I had in mind. Wouldn't it be as simple as adjusting the `CLASSPATH` environment variable in the run script ?
As I've checked the `kafka_connect_run.sh` script has clause like this

```bash
# enabling Tracing agent (initializes tracing) as Java agent
if [ "$STRIMZI_TRACING" = "jaeger" ] || [ "$STRIMZI_TRACING" = "opentelemetry" ]; then
    KAFKA_OPTS="$KAFKA_OPTS -javaagent:$(ls "$KAFKA_HOME"/libs/tracing-agent*.jar)=$STRIMZI_TRACING"
    export KAFKA_OPTS
    if [ "$STRIMZI_TRACING" = "opentelemetry" ] && [ -z "$OTEL_TRACES_EXPORTER" ]; then
      # auto-set OTLP exporter
      export OTEL_TRACES_EXPORTER="otlp"
    fi
fi
```
Perhaps you simple extend it with `CLASSPATH="${CLASSPATH}:/opt/opentelemery/*.jar" ?
And use this path for the `-javaagenet:` part.

**Comment by scholzj** [comment]
I do not really know how simple it would be to implement #7105 to be honest. I never had time to think about it beyond the original idea.

**Comment by scholzj** [comment]
Discussed on a community call on 22.8.2024: On the Strimzi side, this would be addressed in #7105. We should close this issue and keep #7105 for tracking.

This could be also alternatively addressed in Kafka otself by reducing the classpath that the plugin uses because the same issue can also happen with libraries that are part of Apache Kafka itself and there is no way Strimzi can deal with that. That has to be don in Apache Kafka.
